www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxUniqueDefaultServerConstraint.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+* Copyright (C) 2021 Markus Peter mpeter at one-it.de
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice,
+*    this list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in the
+*    documentation and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+* INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+* AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+* OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+namespace OPNsense\Base\Constraints;
+
+use Phalcon\Messages\Message;
+
+/**
+ * a very specific nginx check - not reusable
+ *
+ * Class NgxUniqueDefaultServerConstraint
+ * @package OPNsense\Nginx\Constraints
+ */
+class NgxUniqueDefaultServerConstraint extends BaseConstraint
+{
+    public function validate(\Phalcon\Validation $validator, $attribute): bool
+    {
+        $node = $this->getOption('node');
+        if ($node) {
+			$httpServerNode = $node->getParentNode();
+			$defaultServerNode = $httpServerNode->getChild("default_server");
+            if (!$this->isEmpty($defaultServerNode))
+            {
+				$myUUID = $httpServerNode->getAttribute("uuid");
+				$myListenHTTPAddress = $httpServerNode->getChild("listen_http_address");
+				
+				$httpServersNode = $httpServerNode->getParentNode();
+				
+				$httpServers = $httpServersNode->getChildren();
+
+				$msg = "";				
+				foreach ($httpServers as $httpServer)
+				{
+					$uuid = $httpServer->getAttribute("uuid");
+					if ($uuid != $myUUID)
+					{
+						$defaultServerNode = $httpServer->getChild("default_server");
+						if (!$this->isEmpty($defaultServerNode))
+						{
+							$listenHTTPAddressNode = $httpServer->getChild("listen_http_address");
+							if ($this->compareListenAddresses($myListenHTTPAddress, $listenHTTPAddressNode, $msg))
+							{
+								$msg = "There can only be one Default Server on each listening address: " . $msg . " conflict.";
```

Comment:
this needs to be capable of i18n. We do use the gettext function for that.
Example:
https://github.com/opnsense/plugins/blob/master/www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NaxsiIdentifierConstraint.php#L81

Since you are using a variable inside, you have to replace the placeholder. 
Since your string contains a placeholder, you need to make it work with the `sprintf` function.

For example, it may look like this in the resulting PO file (this is from german (de-DE.po)):
```
#, php-format
msgid ""
"%s is not a valid start destination port. It must be a port alias or integer "
"between 1 and 65535."
msgstr ""
"%s ist kein gültiger Start-Zielport. Es muss ein Portalias oder eine Zahl "
"zwischen 1 und 65535 sein."
```

Reply:
done